Lib/enum.py

```diff
@@ -1017,86 +1214,60 @@ def __xor__(self, other):
         return self.__class__(self._value_ ^ other._value_)
 
     def __invert__(self):
-        members, uncovered = _decompose(self.__class__, self._value_)
-        inverted = self.__class__(0)
-        for m in self.__class__:
-            if m not in members and not (m._value_ & self._value_):
-                inverted = inverted | m
-        return self.__class__(inverted)
+        if self._inverted_ is None:
+            if self._boundary_ is KEEP:
+                # use all bits
+                self._inverted_ = self.__class__(~self._value_)
+            else:
+                # get flags not in this member
+                self._inverted_ = self.__class__(reduce(
+                        _or_,
+                        [m._value_ for m in self.__class__ if m not in self],
+                        0
+                        ))
```

Comment:
it should just be a bit operation... like `flag_mask ^ self._value`

Reply:
Change made.